pkg/client/client.go

```diff
@@ -974,7 +974,21 @@ func (c *Client) WaitForDaemonSetRollout(ds *appsv1.DaemonSet) error {
 				d.Status.DesiredNumberScheduled, d.Status.UpdatedNumberScheduled)
 			return false, nil
 		}
-		if d.Status.NumberUnavailable != 0 {
+
+		var nodeNotReadyCount int32
+		nodeList, err := c.kclient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		nodes := nodeList.Items
+		for _, node := range nodes {
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == v1.NodeReady {
+					if condition.Status != v1.ConditionTrue {
+						nodeNotReadyCount++
+					}
+				}
+			}
+		}
+
+		if d.Status.NumberUnavailable != 0 && d.Status.NumberUnavailable != nodeNotReadyCount {
```

Comment:
another way to look at it would be to compare the number of ready nodes against `NumberAvailable`. If there's a difference, CMO could log something like `expected %d ready pods for %q daemonset, got only %d`

Reply:
Done